src/snapred/backend/recipe/algorithm/PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -161,20 +161,11 @@ def loadNeededLogs(self, ws_name: str, instrumentState: InstrumentState):
         # NOTE after adding the logs, it is necessary to update the instrument to
         #  factor in these new parameters, or else calculations will be inconsistent.
         #  This is done with a call to `ws->populateInstrumentParameters()` from within mantid.
-        # TODO use this sample log, after Mantid PR 36524 has gone into main
-        # https://github.com/mantidproject/mantid/pull/36524
-        # self.mantidSnapper.AddSampleLog(
-        #     "Update the instrument",
-        #     Workspace=ws_name,
-        #     LogName="update_instrument",
-        #     UpdateInstrumentParameters=True,
-        # )
-        # TODO remove the below after uncommenting above.
-        minimalXML = "<parameter-file>></parameter-file>"
-        self.mantidSnapper.LoadParameterFile(
-            "Calling an algorithm that includes a populateInstrumentParamters and no file load",
+        self.mantidSnapper.AddSampleLog(
+            "Update the instrument",
             Workspace=ws_name,
-            ParameterXML=minimalXML,
+            LogName="update_instrument",
+            UpdateInstrumentParameters=True,
```

Comment:
You should get rid of these lines and put the `UpdateInstrumentParameters=True` in the `AddSampleLog` on L153. Calling it an extra time with a log that doesn't belong is silly.

Reply:
It doesn't need to update the instrument parameters until after all four sample logs have been added, so it's kind of a waste of compute cycles to have it update the instrument within the for loop.